applications/nrf_desktop/doc/usb_state.rst

```diff
@@ -6,3 +6,66 @@ USB state module
 The ``usb_state`` module is responsible for tracking the USB connection.
 It is also responsible for the data transmission that application does on this
 device.
+
+Module Events
+*************
+
++--------------------+----------------------------+-------------+-----------------------------------+------------------+
+| Source Module      | Input Event                | This Module | Output Event                      | Sink Module      |
++====================+============================+=============+===================================+==================+
++                    + ``module_state_event``     +             + ``module_state_event``            +                  +
++--------------------+----------------------------+             +-----------------------------------+------------------+
++ ``hid_state``      + ``hid_report_event``       +             + ``hid_report_sent``               + ``hid_state``    +
++--------------------+----------------------------+             +-----------------------------------+                  +
++                    + ``config_event``           +             + ``hid_report_subscription_event`` +                  +
++--------------------+----------------------------+             +-----------------------------------+------------------+
++                    + ``config_fetch_event``     +             + ``usb_state_event``               +                  +
++--------------------+----------------------------+             +-----------------------------------+------------------+
++                    + ``config_forwarder_event`` +             +                                   +                  +
++--------------------+----------------------------+-------------+-----------------------------------+------------------+
+
+
+Configuration
+*************
+
+The ``usb_state`` is enabled by selecting ``CONFIG_DESKTOP_USB_ENABLE``.
+It depends on :option:`CONFIG_USB_DEVICE_HID`.
+
+When enabling the USB support for the device, set the following generic options:
+
+ * :option:`CONFIG_USB_DEVICE_MANUFACTURER` - the device manufacturer's name
+ * :option:`CONFIG_USB_DEVICE_PRODUCT` - the device's product name
+ * :option:`CONFIG_USB_DEVICE_VID` - the device's vendor ID number
+ * :option:`CONFIG_USB_DEVICE_PID` - the device`s product ID number
+
+For low latency devices, make sure that the device requests a polling rate of 1 ms by setting :option:`CONFIG_USB_HID_POLL_INTERVAL_MS` to 1.
+
+If the device is meant to support the boot protocol, select :option:`CONFIG_USB_HID_BOOT_PROTOCOL` and make sure :option:`CONFIG_USB_HID_PROTOCOL_CODE` chooses between a mouse or a keyboard code.
+
+Implementation details
+**********************
+
+The ``usb_state`` module registers the HID-class USB device and initializes the USB stack.
```

Comment:
Initializes USB HID and enables the USB.
USB device stack is actually initialized and configured with `SYS_INIT` macro by Zephyr.

Reply:
Not anymore, this was changed few months back.

Reply:
To be exact: we register the HID USB device, initialize the USB HID device and enable the USB.

The USB stack is still configured in the `SYS_INIT` by the `usb_device_init` - the function configures the USB device stack based on USB descriptor.